VDA-StorageNode-Registry/contracts/IStorageNodeRegistry.sol

```diff
@@ -0,0 +1,215 @@
+     */
+    event AddNode(
+        address indexed didAddress, 
+        string indexed endpointUri,
+        string countryCode,
+        string regionCode,
+        uint datacenterId,
+        int lat,
+        int long
+    );
+
+    /**
+     * @notice Emitted when a removing node is requested
+     * @param didAddress DID address that is to be removed from the network
+     * @param unregisterDateTime The unix timestamp of when the storage node should no logner be available for selection.
+        Must be at leaset 28 dayse in the future from calling function point
+     */
+    event RemoveNodeStart(address indexed didAddress, uint unregisterDateTime);
+
+    /**
+     * @notice Emitted when a removing node is completed
+     * @param didAddress DID address that is to be removed from the network
+     */
+    event RemoveNodeComplete(address indexed didAddress);
+
+    /**
+     * @notice Add a data center to the network. `id` will be auto-incremented.
+     * @dev Only the contract owner can call this function
+     * @param data Datacenter info
+     * @return datacenterId Created datacetnerId
+     */
+    function addDatacenter(Datacenter calldata data) external payable returns(uint);
+
+    /**
+     * @notice Remove a data center
+     * @dev Only the contract owner call call this function.
+     *  Will only remove the data center if there are no storage nodes using this datacenterId
+     * @param datacenterId datacenterId created by `addDatacenter()` function
+     */
+    function removeDatacenter(uint datacenterId) external payable;
+
+    /**
+     * @notice Return an array of `Datacenter` structs for given array of datacenterIds
+     * @param ids Array of datacenterIds
+     * @return Datacenter[] Array of `Datacenter` structs 
+     */
+    function getDatacenters(uint[] calldata ids) external view returns(Datacenter[] memory);
+
+    /**
+     * @notice Return an array of `Datacenter` structs for country code
+     * @param countryCode Unique two-character string code
+     * @return Datacenter[] Array of `Datacenter` structs 
+     */
+    function getDataCentersByCountry(string calldata countryCode) external view returns(Datacenter[] memory);
+
+    /**
+     * @notice Return an array of `Datacenter` structs for region
+     * @param regionCode Unique region string code
+     * @return Datacenter[] Array of `Datacenter` structs 
+     */
+    function getDataCentersByRegion(string calldata regionCode) external view returns(Datacenter[] memory);
+    
+    /**
+     * @notice Registers a new endpoint on the network
+     * @dev A did can register only one storage-node
+     * @param nodeInfo Node information to be added
+     * @param requestSignature The request parameters signed by the `didAddress` private key
+     * @param requestProof Used to verify request
+     * @param authSignature Signature signed by a trusted signer
```

Comment:
Please add this parameter and a description of how it's used to the VIP-10

Reply:
Updated : https://github.com/verida/VIPs/pull/5